demisto_sdk/commands/test_content/TestContentClasses.py

```diff
@@ -3244,3 +3251,99 @@ def __str__(self):
+        """
+        logging.info(
+            f"Starting update_content_status with {len(failed_tests)} failed tests and "
+            f"{len(successful_tests)} successful tests."
+        )
+
+        self._load_content_status()
+        self._initialize_content_status_keys()
+
+        self._update_playbooks("failed_playbooks", failed_tests)
+        self._update_playbooks("successful_playbooks", successful_tests)
+
+        self._save_content_status()
+
+    def _load_content_status(self) -> None:
+        """
+        Attempts to load the content status from the file. If the file doesn't exist or is invalid, initializes an empty status.
+        """
+        if Path.exists(self.content_status_path):
+            logging.info(f"Content status file exists at {self.content_status_path}")
+            try:
+                with open(self.content_status_path, "r") as content_file:
+                    self.content_status = json.load(content_file)
+                    logging.info(f"Loaded content status: {self.content_status}")
+            except json.JSONDecodeError as e:
+                logging.error(
+                    f"JSON decode error while loading content_status.json: {e}"
+                )
+                self.content_status = {}
+        else:
+            logging.info(
+                f"Initializing empty content status at {self.content_status_path}"
+            )
+            self.content_status = {}
+
+    def _initialize_content_status_keys(self) -> None:
+        """
+        Ensures that the 'failed_playbooks' and 'successful_playbooks' keys are initialized in the content status.
+        """
+        for key in ["failed_playbooks", "successful_playbooks"]:
+            if key not in self.content_status:
+                logging.info(
+                    f"'{key}' key not in content_status. Initializing to empty list."
+                )
+                self.content_status[key] = []
+
+    def _update_playbooks(self, key: str, tests: List[str]) -> None:
+        """
+        Updates the given key in the content status dictionary with the provided list of tests, ensuring no duplicates.
+
+        Args:
+            key (str): The key ('failed_playbooks' or 'successful_playbooks') to update.
+            tests (List[str]): The list of tests (either successful or failed) to add to the content status.
+        """
+        current_playbooks = self.content_status.get(key, [])
+        new_playbooks = [test for test in tests if test not in current_playbooks]
+        if new_playbooks:
+            logging.info(f"Adding {len(new_playbooks)} new {key}: {new_playbooks}")
+            current_playbooks.extend(new_playbooks)
+        else:
+            logging.info(f"No new {key} to add.")
+
+    def _save_content_status(self) -> None:
+        """
+        Saves the updated content status back to the specified file path.
+        """
+        os.makedirs(os.path.dirname(self.content_status_path), exist_ok=True)
```

Comment:
```suggestion
        self.content_status_path.mkdir(exist_ok=True)
```
Unless you mean to create a path in cwd named after that folder, then 
```suggestion
        Path(self.content_status_path.name).mkdir(exists_ok=True)
```